shared/fs/index.js

```diff
@@ -1,30 +1,110 @@
 // @flow
-import * as React from 'react'
-import {globalStyles, globalMargins} from '../styles'
-import {Box, Button, Text, ButtonBar} from '../common-adapters'
-
-type Props = {
-  counter: number,
-  you: ?string,
-  increase: () => void,
-  increase10: () => void,
+import React, {PureComponent} from 'react'
+import {FolderVisibility} from '../constants/types/fs'
+import {globalStyles, globalColors, globalMargins} from '../styles'
+import {isMobile} from '../constants/platform'
+import {Box, Button, ClickableBox, Icon, List, Text} from '../common-adapters'
+import {type IconType} from '../common-adapters/icon'
+
+const stylesCommonCore = {
+  alignItems: 'center',
+  borderBottomColor: globalColors.black_05,
+  borderBottomWidth: 1,
+  justifyContent: 'center',
 }
 
-const Fs = ({counter, you, increase, increase10}: Props) => (
-  <Box style={containerStyle}>
-    <Text type="Header">Hi {you}!</Text>
-    <Text type="BodySemibold">Count: {counter}</Text>
-    <ButtonBar align="flex-start">
-      <Button type="Primary" onClick={increase} label="Up by 1" />
-      <Button type="Primary" onClick={increase10} label="Up by 10" />
-    </ButtonBar>
-  </Box>
-)
+const stylesCommonRow = {
+  ...globalStyles.flexBoxRow,
+  ...stylesCommonCore,
+  minHeight: isMobile ? 64 : 48,
+  padding: 8,
+}
 
-const containerStyle = {
+const stylesContainer = {
   ...globalStyles.flexBoxColumn,
+  ...globalStyles.fullHeight,
   flex: 1,
-  padding: globalMargins.xlarge,
 }
 
+const rootFolders = [
+  'private',
+  'public',
+  'team',
+]
+
+type FolderHeaderProps = {
+  title: string,
+}
+
+type FileRowProps = {
+  path: string,
+  icon: IconType,
+  showFileData: () => void,
+}
+
+type FolderProps = {
+  path: string,
+  visibility: FolderVisibility,
+  items: Array<string>,
+}
+
+const FolderHeader = ({title}: FolderHeaderProps) => (
+  <Box>
+    <Box style={{...stylesCommonRow, alignItems: 'center', borderBottomWidth: 0}}>
+      <Text type="HeaderBig" style={{padding: globalMargins.xtiny}}>
+        {title}
+      </Text>
+    </Box>
+  </Box>
+)
+
+
+const FileRow = ({path, icon, showFileData}: FileRowProps) => (
+  <ClickableBox onClick={showFileData} style={{...stylesCommonRow}}>
+    <Box key={path} style={{...globalStyles.flexBoxRow, alignItems: 'center', flex: 1}}>
+      <Icon type={icon} style={{marginRight: 16}} />
+      <Box style={{...globalStyles.flexBoxColumn, flex: 1, justifyContent: 'flex-start'}}>
+        <Text type="Body" style={{flex: 1}}>
+          {path}
+        </Text>
+      </Box>
+    </Box>
+  </ClickableBox>
+)
+
+class Folder extends PureComponent<FolderProps> {
+  _renderRow(index, item) {
```

Comment:
Just want to point out a subtlety about member methods. This works fine in this instance but note this is *not* a bound method. 

aka. if this method referred to `this.myValue` the 'this' pointer could be undefined.

aka
```
_renderRow(index, item) {
  console.log(this.myValue); 
}
```

will work if you do
```
const folder = new Folder(...)
folder._renderRow(0, 0)
```

but if you do
```
const folder = new Folder(...)
const whoops = folder._renderRow
whoops(0, 0)
```

that will fail. While this seems like a contrived example this is exactly what we're doing passing this._renderRow to `<List items={items} renderItem={this._renderRow} />`

There are a couple of solutions to this (bind, etc) but the simplest and best react wise is to instead make the member an arrow function so 'this' is closed over and automatically bound

tldr;

```
_renderRow = (index, item) => {
```

does not have this problem

Reply:
Does it mean you could do `whoops(folder, 0, 0)` which would be equivalent to `folder._renderRow(0,0)`?